audit/entry_formatter.go

```diff
@@ -107,33 +119,36 @@ func (f *EntryFormatter) Process(ctx context.Context, e *eventlogger.Event) (*ev
 		data.Request.Headers = adjustedHeaders
 	}
 
+	// Using any as we have two different types that we can get back from either
+	// FormatRequest or FormatResponse.
+	var entry any
 	var result []byte
 
 	switch a.Subtype {
 	case RequestType:
-		entry, err := f.FormatRequest(ctx, data)
-		if err != nil {
-			return nil, fmt.Errorf("%s: unable to parse request from audit event: %w", op, err)
-		}
-
-		result, err = jsonutil.EncodeJSON(entry)
-		if err != nil {
-			return nil, fmt.Errorf("%s: unable to format request: %w", op, err)
-		}
+		entry, err = f.FormatRequest(ctx, data)
 	case ResponseType:
-		entry, err := f.FormatResponse(ctx, data)
-		if err != nil {
-			return nil, fmt.Errorf("%s: unable to parse response from audit event: %w", op, err)
-		}
-
-		result, err = jsonutil.EncodeJSON(entry)
-		if err != nil {
-			return nil, fmt.Errorf("%s: unable to format response: %w", op, err)
-		}
+		entry, err = f.FormatResponse(ctx, data)
 	default:
 		return nil, fmt.Errorf("%s: unknown audit event subtype: %q", op, a.Subtype)
 	}
 
+	if err != nil {
+		return nil, fmt.Errorf("%s: unable to parse %s from audit event: %w", op, a.Subtype.HumanReadableString(), err)
+	}
+
+	// Exclude data from the RequestEntry/ResponseEntry.
+	m, err := f.excludeFields(entry)
+	if err != nil {
+		return nil, fmt.Errorf("%s: unable to exclude audit data from %s: %w", op, a.Subtype.HumanReadableString(), err)
+	}
+
+	// Convert map m into a slice of bytes
+	result, err = jsonutil.EncodeJSON(m)
```

Comment:
As per my comment inside exclude fields, if there are no exclusions, we can still do `jsonutil.EncodeJSON(entry)` like we used to have above, right?